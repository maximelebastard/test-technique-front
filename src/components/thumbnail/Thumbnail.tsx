import React from "react";
import styles from "./Thumbnail.module.scss";

type ThumbnailProps = {
  title: string;
  posterUrl?: string;
  onClick?: () => void;
  selected: boolean;
};

const Thumbnail = React.forwardRef<HTMLDivElement, ThumbnailProps>(
  function Thumbnail(props, ref) {
    const { title, selected, posterUrl } = props;

    return (
      <div
        ref={ref}
        onClick={() => props.onClick && props.onClick()}
        className={[
          styles.MovieItem,
          selected ? styles.selected : undefined
        ].join(" ")}
      >
        <img
          crossOrigin=""
          alt={title}
          className={styles.poster}
          src={posterUrl}
        />
        <div className={styles.title}>{title}</div>
      </div>
    );
  }
);

export default Thumbnail;
