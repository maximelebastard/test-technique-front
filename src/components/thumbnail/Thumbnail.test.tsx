import React from "react";
import { shallow, render } from "enzyme";
import Thumbnail from "./Thumbnail";

it("renders without crashing", () => {
  shallow(
    <Thumbnail
      selected={false}
      title={"Back to the Future"}
      posterUrl="http://some.url/bttf.jpg"
    />
  );
});

it("matches the snapshot", () => {
  expect(
    render(
      <Thumbnail
        selected={true}
        title={"Back to the Future"}
        posterUrl="http://some.url/bttf.jpg"
      />
    ).text()
  ).toMatchSnapshot();
});
