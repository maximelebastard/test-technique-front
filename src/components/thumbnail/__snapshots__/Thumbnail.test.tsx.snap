// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `"Back to the Future"`;
